### ISSUE TITLE
Helping Folks Get Started with Node.js through Social 

### ISSUE THREAD

**Issue by ZibbyKeaton** [nodejs/education#34]
TL;DR 
seeking submissions for:
-  weekly tip per to help folks better understand Node.js (focused on those new to Node)
-  content related to communities and classes available to help folks learn Node.js
- bloggers/commentators that you think do a great job of explaining and teaching Node.js

Hey All - For those of you that don't know me, I work for the Linux Foundation and help run marketing/public relations for the Node.js Foundation. I also help build content for the Node.js Foundation's social handles (Twitter, Facebook, LinkedIn, and Medium).

Given there are a ton of folks just learning Node.js (whether they are new developers or developers in transition), I thought it would be helpful to increase some of our social posts to be focused on resources to help folks get started in Node.js and provide them with short tips and best practices when working with Node.js.

Initially, I think it would be nice to do a tip per week to help folks better understand Node.js. For example, we could post something like "Tip of the Week with Node.js: Lock Your Dependencies" or link to a great online resource available to new developers or developers that are just transitioning to Node.js

Beyond this, I would also like to add more content related to communities and classes available to help folks learn Node.js as well as any bloggers/commentators that focus on the educational side of learning Node.js/JavaScript, etc.

If you have any ideas on tips or resources that you think we should highlight, please share. 

Thanks for your insight and expertise!

**Comment by DiegoZoracKy** [comment]
Hi @ZibbyKeaton, just a question to confirm my understanding. A content focused on **Promises**, which is related to JavaScript in general and not solely about Node.js, would be included in this scope? For example, when talking about **Promises** one could include a topic on the new **util.promisify()**, which is from the Node.js world, but let's suppose that this is not the case. The article/content is about best practices when using Promises, and the Node.js environment is not the main subject, even though it is being used for code examples. 

From my point of view, when introducing Node.js to those who are new to it, but are comfortable with other languages, there is a bunch to be explained which is not just diving into Node.js itself. The transition from a "sync" way of thinking to an "async" one, for example, is a very important topic. One that could end up being written in an article without going much deep into Node.js methods/API in a first moment.

**Comment by ZibbyKeaton** [comment]
@DiegoZoracKy really great point here. I think the types of articles you are referencing above will very much be helpful and something that we will want to highlight even if we are not directly talking about Node.js.

**Comment by DiegoZoracKy** [comment]
Hi @ZibbyKeaton. I'm back just to share some thoughts on this topic that went through my mind today. I see more than one kind of public within "focused on those new to Node" and "help folks learn Node.js". One group is not into Node.js yet, but is probably seeking to know more about it before diving in. Other group is composed of folks who are already trying Node.js, but at their first steps, are reading content about it to go further. And then there is another one, with people who are already breathing Node.js, but are still constantly studying and reading about Node.js to keep improving themselves. Having that in mind i can see contents written on the following topics, ranging from novice to expert levels:

**Internal Architecture**

Contents about Node.js engine, examples:

*How does Node.js handles IO*
*Event Loop*
*Differences between single thread and multi thread apps*
*Async vs Sync (architecture level)*

**Culture**

Contents on the culture and philosophies that comes to us when working with Node.js. Could be talking about the **modular** way of thinking, which turns to be possible, and reinforced, by the **dependency** management system that makes it simple. Semantic Versioning (**semver**). The culture that is brought by the **community**, which goes beyond the **Node.js** itself (modules, ways of working, agreements, boundaries pushed, etc.).

**JavaScript**

JavaScript!  

***Programming in Node.js**

Examples, ideas, ways of writing code and structure modules in Node.js.

**Node.js Native API**

Contents on Node.js Native APIs. Use cases, examples, etc.

**Node.js Modules**

Contents on Node.js Modules published by the **community**. I believe this one would be really great. Node.js by itself is a really great engine, but the community adds to it a real huge value. We could have contents on specific modules. How some modules are shaping the way we work, and bringing to us ways of following good practices of software development in general. 

***"Mini"* Case Studies**

Developers and companies telling how Node.js was used to accomplish a specific mission. Not just on the web, which is usually the focus, but about any kind of issue. Like automating internal tasks, complex file share operations, IoT / Single Board Computers, etc.

Now, finishing writing this comment, I realized that there are already contents on the web written on some of these topics. Maybe would be a good idea to prepare an aggregator, properly curated, listing some contents already written out there. Instead of wait and be dependent just on new contents to be written.

**Comment by ZibbyKeaton** [comment]
@DiegoZoracKy thank you, your input is really, really helpful. Agree, there's a lot of great stuff already out there that I can pull from and this was my initial intention (to share great and helpful content). If you see anything, I would love for you to share it with this thread. This also gives me a good base on what to look for as well when I'm doing my own research and finding items that will be interesting and helpful for the variety of folks learning Node.js. Very much appreciate you taking the time. Thank you again.
